src/main/java/com/mraof/minestuck/entity/consort/MessageType.java

```diff
@@ -1036,30 +1039,41 @@ public MutableComponent getMessage(ConsortEntity consort, ServerPlayer player, S
 						addTo(chainIdentifier, conditionedMessage.getString()));
 			
 			boolean hasItem = false;
+			List<ItemStack> stackListFromTag = new ArrayList<>();
+			Registry.ITEM.getTagOrEmpty(itemTag).forEach(itemHolder -> stackListFromTag.add(new ItemStack(itemHolder)));
+			
 			if(random || repeat && nbt.contains(this.getString()))
 			{
-				int index;
+				String string;
 				if(nbt.contains(this.getString()))
-					index = nbt.getInt(this.getString());
+				{
+					string = nbt.getString(this.getString());
+				}
 				else
 				{
-					index = consort.level.random.nextInt(possibleItems.size());
-					nbt.putInt(this.getString(), index);
+					int index = consort.level.random.nextInt(stackListFromTag.size());
+					ItemStack randomStack = stackListFromTag.get(index);
+					string = randomStack.getItem().getRegistryName().toString();
+					nbt.putString(this.getString(), string);
 				}
 				
-				ItemStack stack = possibleItems.get(index);
-				nbt.put(this.getString() + ".item", stack.save(new CompoundTag()));
-				
-				hasItem = lookFor(stack, player);
+				Optional<Item> optionalItem = Registry.ITEM.getOptional(new ResourceLocation(string));
+				if(optionalItem.isPresent())
+				{
+					ItemStack stack = new ItemStack(optionalItem.get());
+					nbt.put(this.getString() + ".item", stack.save(new CompoundTag()));
+					
+					hasItem = lookFor(stack, player);
+				}
 			} else
 			{
-				List<ItemStack> list = new ArrayList<>(possibleItems);
+				List<ItemStack> list = new ArrayList<>(stackListFromTag);
 				while (!list.isEmpty())
 				{
 					ItemStack stack = list.remove(consort.level.random.nextInt(list.size()));
 					if(lookFor(stack, player))
 					{
-						nbt.putInt(this.getString(), possibleItems.indexOf(stack));
+						nbt.putInt(this.getString(), stackListFromTag.indexOf(stack));
```

Comment:
This should put a string to match the changes above.

Reply:
oops missed that